File: frontend/src/types/invoice.ts
// 上传成功后后端返回的抽取字段结构。
export interface InvoiceExtracted {
  amount: number | null;
  date: string | null;
  seller_name: string | null;
  purpose: string | null;
  invoice_number: string | null;
  title: string | null;
  tax_id: string | null;
  item_name: string | null;
}

// 上传接口响应：包含记录 ID、归档文件名与覆盖标记。
export interface UploadInvoiceResponse {
  id: number;
  file_name: string;
  uploader_name: string | null;
  replaced: boolean;
  message: string | null;
  extracted: InvoiceExtracted;
}

// 会计列表页面使用的行数据结构。
export interface InvoiceRecord {
  id: number;
  file_name: string;
  uploader_name: string | null;
  amount: number | null;
  invoice_date: string | null;
  seller_name: string | null;
  purpose: string | null;
  invoice_number: string | null;
  title: string | null;
  tax_id: string | null;
  item_name: string | null;
  created_at: string;
  approval_status: "draft" | "pending" | "approved" | "rejected";
}

// 发票详情结构：含文件链接、预览图与原始文本。
export interface InvoiceDetail {
  id: number;
  file_name: string;
  uploader_name: string | null;
  amount: number | null;
  invoice_date: string | null;
  seller_name: string | null;
  purpose: string | null;
  invoice_number: string | null;
  tax_id: string | null;
  source_file_name: string | null;
  archived_file_name: string | null;
  source_file_url: string | null;
  archived_file_url: string | null;
  source_preview_image_url: string | null;
  archive_preview_image_url: string | null;
  preview_image_url: string | null;
  raw_text: string | null;
  created_at: string;
  approval_status: "draft" | "pending" | "approved" | "rejected";
  approval_comment: string | null;
  approver_name: string | null;
  approved_at: string | null;
}

File: backend/app/schemas/invoice.py
from pydantic import BaseModel, ConfigDict


# LLM 与规则兜底后的字段结构。
class InvoiceExtractedData(BaseModel):
    amount: float | None = None
    date: str | None = None
    seller_name: str | None = None
    purpose: str | None = None
    invoice_number: str | None = None
    tax_id: str | None = None
    title: str | None = None
    item_name: str | None = None


class InvoiceCreateResponse(BaseModel):
    id: int
    file_name: str
    uploader_name: str | None = None
    replaced: bool = False
    message: str | None = None
    extracted: InvoiceExtractedData


class InvoiceListItem(BaseModel):
    id: int
    file_name: str
    uploader_name: str | None
    amount: float | None
    invoice_date: str | None
    seller_name: str | None
    purpose: str | None
    invoice_number: str | None
    title: str | None
    tax_id: str | None
    item_name: str | None
    created_at: str
    approval_status: str = "pending"

    model_config = ConfigDict(from_attributes=True)


class InvoiceDetailResponse(BaseModel):
    id: int
    file_name: str
    uploader_name: str | None
    amount: float | None
    invoice_date: str | None
    seller_name: str | None
    purpose: str | None
    invoice_number: str | None
    tax_id: str | None
    source_file_name: str | None
    archived_file_name: str | None
    source_file_url: str | None
    archived_file_url: str | None
    source_preview_image_url: str | None
    archive_preview_image_url: str | None
    preview_image_url: str | None
    raw_text: str | None
    created_at: str
    approval_status: str = "pending"
    approval_comment: str | None = None
    approver_name: str | None = None
    approved_at: str | None = None


class ApprovalRequest(BaseModel):
    status: str
    comment: str | None = None
    approver_name: str


class ApprovalResponse(BaseModel):
    id: int
    approval_status: str
    approval_comment: str | None
    approver_name: str | None
    approved_at: str | None
